mayflower/src/main/java/io/github/madethoughts/mayflower/configuration/PluginConfig.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.madethoughts.mayflower.configuration;
+
+import io.micronaut.context.annotation.AliasFor;
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.core.annotation.AccessorsStyle;
+import io.micronaut.core.version.annotation.Version;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ This annotation indicates, that this configuration is the plugin's config. The values for this config will also be
+ loaded from the "config.yml" file in the plugins' directory. Also, all entries of this configuration need a default
+ value, for that {@link Default} can be used. The corresponding "config.yml" will have default values based on them.
+ <p>
+ All {@link PluginConfig} implementations must have a {@link Version} annotation. The version format is "minor.major"
```

Comment:
`minor.major`? Shouldn't it be `major.minor`?
```suggestion
 All {@link PluginConfig} implementations must have a {@link Version} annotation. The version format is "major.minor"
```